src/v2/auth.rs

```diff
@@ -185,3 +176,54 @@ impl Client {
         Box::new(fres)
     }
 }
+
+// This parses a Www-Authenticate header of value Bearer, which could look something like this:
+// `Www-Authenticate: Bearer realm="https://sat-r220-02.lab.eng.rdu2.redhat.com/v2/token",service="sat-r220-02.lab.eng.rdu2.redhat.com",scope="repository:registry:pull,push"`
+// We are interested in the realm and service keys
+fn parse_hdr_bearer(input: &str) -> Result<(String, Option<&str>)> {
+    let mut auth_ep = "".to_string();
+    let mut service = None;
+
+    let re = regex::Regex::new(r#"(([a-z]+)="([^"]*)")"#)?;
+    for capture in re.captures_iter(input) {
+        match (
+            // The indices for this capture are as follows:
+            // 0: full match
+            // 1: outer group match
+            // 2: first nested group match
+            // 3: second nested group match
+            // Hence, we are interested in the sub-group matches i.e. in 2 and 3.
```

Comment:
Thanks for the detailed comment. Just personal taste, but the match with nested comment looks funny.
What about naming those two variables (key+value?) and moving them plus comments outside of the match body?